### PR TITLE
Add Tair to examples of vector database

### PR DESCRIPTION
**Pull request by dongqqcom**
Hi there,

I'm excited to open this PR to add an example of support for using a database `Tair` as a datastore. 

[Tair](https://www.alibabacloud.com/help/en/tair/latest/what-is-tair) is a cloud native in-memory database service that is developed by Alibaba Cloud. It is compatible with open source Redis and provides a variety of data models and enterprise-class capabilities to support real-time online scenarios. Tair also introduces persistent memory-optimized instances that are based on the new non-volatile memory (NVM) storage medium. These instances can reduce costs by 30%, ensure data persistence, and provide almost the same performance as in-memory databases. Tair has been widely used in areas such as government affairs, finance, manufacturing, healthcare, and pan-Internet to meet their high-speed query and computing requirements.

[Tairvector](https://www.alibabacloud.com/help/en/tair/latest/tairvector) is an in-house data structure that provides high-performance real-time storage and retrieval of vectors. TairVector provides two indexing algorithms: Hierarchical Navigable Small World (HNSW) and Flat Search. Additionally, TairVector supports multiple distance functions, such as Euclidean distance, inner product, and Jaccard distance. Compared with traditional vector retrieval services, TairVector has the following advantages:
- Stores all data in memory and supports real-time index updates to reduce latency of read and write operations.
- Uses an optimized data structure in memory to better utilize storage capacity.
- Functions as an out-of-the-box data structure in a simple and efficient architecture without complex modules or dependencies.

Therefore, `Tair` is a good choice for vector storage.

### CODE REVIEW

**Comment by dongqqcom** [comment]
@ted-at-openai hey can you please take a look at this PR？Thanks!

File: examples/vector_databases/tair/QA_with_Langchain_Tair_and_OpenAI.ipynb [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2023-05-06T10:21:40.974668Z",
+     "start_time": "2023-05-06T10:21:40.845980Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "! export OPENAI_API_KEY=\"your API key\""
```
Comment (simonpfish) [inline examples/vector_databases/tair/QA_with_Langchain_Tair_and_OpenAI.ipynb:80]: We generally don't want to encourage people writing their OpenAI key in a code block. Instead, use something like `getpass`

File: examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+    "Once you get your key, please add it to your environment variables as `OPENAI_API_KEY`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! export OPENAI_API_KEY=\"your API key\""
```
Comment (simonpfish) [inline examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb:90]: same goes here

File: examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The format of url: redis://[[username]:[password]]@localhost:6379/0\n",
+    "! export TAIR_URL=\"your_tair_url\""
```
Comment (simonpfish) [inline examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb:150]: you might also want to query for this url with `getpass`

File: examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+    "\n",
+    "# The file is ~700 MB so this will take some time\n",
+    "wget.download(embeddings_url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The downloaded file has to be then extracted:"
```
Comment (simonpfish) [inline examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb:220]:
```suggestion
    "The downloaded file has to then be extracted:"
```

**Review by simonpfish — CHANGES_REQUESTED** [review]
Thanks for the contribution. Left a couple comments for small things to address

**Comment by dongqqcom** [comment]
你好，我已收到你的邮件，会尽快给你回复。

File: examples/vector_databases/tair/QA_with_Langchain_Tair_and_OpenAI.ipynb (hunk 1 above)
Comment (dongqqcom) [reply examples/vector_databases/tair/QA_with_Langchain_Tair_and_OpenAI.ipynb:80]: Thanks for your suggestions! We will change the OpenAI key input way to getpass.

File: examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb (hunk 3 above)
Comment (dongqqcom) [reply examples/vector_databases/tair/Getting_started_with_Tair_and_OpenAI.ipynb]: We have updated the input of api key by getpass in the latest commit.

**Comment by simonpfish** [comment]
Thank you for the updates! Would it be possible to run the code cells and include the output? We've generally found that it is useful for users to be able to see example outputs without executing the code themselves.

**Comment by dongqqcom** [comment]
> Thank you for the updates! Would it be possible to run the code cells and include the output? We've generally found that it is useful for users to be able to see example outputs without executing the code themselves.

Ok, we will add outputs in a new commit.

**Comment by dongqqcom** [comment]
> Thank you for the updates! Would it be possible to run the code cells and include the output? We've generally found that it is useful for users to be able to see example outputs without executing the code themselves.

Thanks for your suggestions! We have added the output of code cells in the latest commit.

**Review by simonpfish — APPROVED** [review]

**Pull request merged**